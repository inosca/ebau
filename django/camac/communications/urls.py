from rest_framework.routers import SimpleRouter

from . import views

r = SimpleRouter(trailing_slash=False)

r.register(r"communications-topics", views.TopicView, "communications-topic")
r.register(r"communications-messages", views.MessageView, "communications-message")
r.register(
    r"communications-attachments", views.AttachmentView, "communications-attachment"
)

urlpatterns = r.urls
