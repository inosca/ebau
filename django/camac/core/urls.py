from rest_framework.routers import SimpleRouter

from . import views

r = SimpleRouter(trailing_slash=False)

r.register(r"publication-entries", views.PublicationEntryView, "publication")
r.register(r"authorities", views.AuthorityView, "authority")
r.register(r"workflow-entries", views.WorkflowEntryView, "workflow-entry")
r.register(r"resources", views.ResourceView, "resource")
r.register(r"instance-resources", views.InstanceResourceView, "instance-resource")
r.register(r"static-contents", views.StaticContentView, "static-content")
r.register(r"service-contents", views.ServiceContentViewSet, "service-content")

urlpatterns = r.urls
