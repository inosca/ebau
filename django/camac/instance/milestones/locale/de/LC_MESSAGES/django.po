# SOME DESCRIPTIVE TITLE.
# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
# This file is distributed under the same license as the PACKAGE package.
# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
#
msgid ""
msgstr ""
"Project-Id-Version: \n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: 2024-09-11 17:25+0200\n"
"PO-Revision-Date: 2024-08-29 13:33+0200\n"
"Last-Translator: \n"
"Language-Team: \n"
"Language: de\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Plural-Forms: nplurals=2; plural=(n != 1);\n"
"X-Generator: Poedit 3.0.1\n"

#: camac/instance/milestones/serializers.py
msgid "regular process"
msgstr "Ordentliches Verfahren"

#: camac/instance/milestones/serializers.py
msgid "instance submitted"
msgstr "Dossier erfasst"

#: camac/instance/milestones/serializers.py
msgid "Paper submission"
msgstr "Eingang Post"

#: camac/instance/milestones/serializers.py
msgid "Additional demand"
msgstr "Nachforderung Gemeindebaubehörde"

#: camac/instance/milestones/serializers.py
msgid "Check completed"
msgstr "Vollständigkeitsprüfung abgeschlossen"

#: camac/instance/milestones/serializers.py
msgid "Instance complete"
msgstr "Dossier vollständig"

#: camac/instance/milestones/serializers.py
msgid "Review building commission"
msgstr "Prüfung durch Gemeinde"

#: camac/instance/milestones/serializers.py
msgid "Puclication date"
msgstr "Publikationsdatum"

#: camac/instance/milestones/serializers.py
msgid "Receipt of objections"
msgstr "Eingang von Einsprachen"

#: camac/instance/milestones/serializers.py
msgid "Forwarding to KOOR"
msgstr "Weiterleitung an KOOR"

#: camac/instance/milestones/serializers.py
msgid "Start circulation"
msgstr "Start Zirkulation"

#: camac/instance/milestones/serializers.py
msgid "Distribution completed"
msgstr "Zirkulation beendet"

#: camac/instance/milestones/serializers.py
msgid "Statement - preliminary decision"
msgstr "Stellungnahme - Vorentscheid"

#: camac/instance/milestones/serializers.py
msgid "Building decision"
msgstr "Bau- und Einspracheentscheid"

#: camac/instance/milestones/serializers.py
msgid "notice to geometer"
msgstr "Meldung Baubewilligung an Geometer"

#: camac/instance/milestones/serializers.py
msgid "Receipt confirmation of decision documents"
msgstr "Empfangsbestätigung Entscheiddokumente"

#: camac/instance/milestones/serializers.py
msgid "Receipt confirmation of preliminary decision"
msgstr "Empfangsbestätigung Vorentscheid"

#: camac/instance/milestones/serializers.py
msgid "notice to gebaeudeschaetzung"
msgstr "Meldung Baubewilligung an Gebäudeschätzung"

#: camac/instance/milestones/serializers.py
msgid "Building permit valid until"
msgstr "Baubewilligung gültig bis"
