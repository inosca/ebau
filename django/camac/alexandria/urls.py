from alexandria.core import views as alexandria_views
from django.urls import path
from rest_framework.routers import SimpleRouter

from camac.alexandria import views

r = SimpleRouter(trailing_slash=False)

# override the default viewset with our patched versions
# this needs to be updated when new urls are added in alexandria
r.register(r"categories", views.PatchedCategoryViewSet)
r.register(r"documents", views.PatchedDocumentViewSet)
r.register(r"files", views.PatchedFileViewSet)
r.register(r"tags", views.PatchedTagViewSet)
r.register(r"tagsynonymgroups", views.PatchedTagSynonymGroupViewSet)
r.register(r"marks", views.PatchedMarkViewSet)

urlpatterns = r.urls

urlpatterns.append(path("hook", alexandria_views.hook_view, name="hook"))
