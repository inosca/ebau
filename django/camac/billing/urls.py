from django.urls import re_path
from rest_framework.routers import SimpleRouter

from camac.billing import views
from camac.billing.export.views import BillingV2EntryExportView

urlpatterns = [
    re_path(
        r"billing-v2-entries/export$",
        BillingV2EntryExportView.as_view(),
        name="billing-export",
    ),
]

r = SimpleRouter(trailing_slash=False)

r.register(r"billing-v2-entries", views.BillingV2EntryViewset, "billing-v2-entry")

urlpatterns.extend(r.urls)
