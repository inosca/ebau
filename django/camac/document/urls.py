from django.urls import re_path
from rest_framework.routers import SimpleRouter

from . import views


class UnswaggeredAttachmentView(views.AttachmentView):
    swagger_schema = None


class UnswaggeredAttachmentSectionView(views.AttachmentSectionView):
    swagger_schema = None


r = SimpleRouter(trailing_slash=False)

r.register(r"attachments", views.AttachmentView)
r.register(r"attachment-sections", views.AttachmentSectionView)
r.register(r"templates", views.TemplateView)
r.register(r"attachment-download-history", views.AttachmentDownloadHistoryView)

urlpatterns = [
    re_path(
        r"^(?P<path>attachments/files/.+\..+)$",
        views.AttachmentDownloadView.as_view({"get": "retrieve"}),
        name="attachment-download",
    ),
    re_path(
        r"^attachments/files/",
        views.AttachmentDownloadView.as_view({"get": "list"}),
        name="multi-attachment-download",
    ),
]

urlpatterns.extend(r.urls)
