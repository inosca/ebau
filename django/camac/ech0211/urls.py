from django.urls import path, re_path
from django.views.generic import RedirectView

from camac.ech0211.views.kt_bern import (
    ApplicationsView as BEApplicationsView,
    ApplicationView as BEApplicationView,
    EventView as BEEventView,
    MessageView as BEMessageView,
    SendView as BESendView,
)
from camac.ech0211.views.kt_schwyz import (
    ApplicationsView as SZApplicationsView,
    ApplicationView as SZApplicationView,
)


class SZUrlsConf:
    urlpatterns = [
        re_path(
            r"application/(?P<instance_id>\d+)/?$",
            SZApplicationView.as_view({"get": "retrieve"}),
            name="application",
        ),
        re_path(
            r"applications",
            SZApplicationsView.as_view({"get": "list"}),
            name="applications",
        ),
    ]


class BEUrlsConf:
    redirects = {
        "instance/<int:instance_id>/": "/page/index/instance-resource-id/20074/instance-id/%(instance_id)i",
        "ebau-number/<int:instance_id>/": "/taskform/taskform/index/instance-resource-id/12000002/instance-id/%(instance_id)i",
        "claim/<int:instance_id>/": "/claim/claim/index/instance-resource-id/150000/instance-id/%(instance_id)i",
        "dossier-check/<int:instance_id>/": "/page/index/instance-resource-id/150009/instance-id/%(instance_id)i",
        "revision-history/<int:instance_id>/": "/revisionhistory/revisionhistory/index/instance-resource-id/150004/instance-id/%(instance_id)i",
    }

    urlpatterns = [
        re_path(
            r"application/(?P<instance_id>\d+)/?$",
            BEApplicationView.as_view({"get": "retrieve"}),
            name="application",
        ),
        re_path(
            r"applications",
            BEApplicationsView.as_view({"get": "list"}),
            name="applications",
        ),
        re_path(
            r"message/$", BEMessageView.as_view({"get": "retrieve"}), name="message"
        ),
        re_path(
            r"event/(?P<instance_id>(\d+))/(?P<event_type>(\w+))/?$",
            BEEventView.as_view({"post": "create"}),
            name="event",
        ),
        re_path(r"send/$", BESendView.as_view({"post": "create"}), name="send"),
    ] + [path(key, RedirectView.as_view(url=url)) for key, url in redirects.items()]
