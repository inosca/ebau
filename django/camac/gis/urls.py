from django.urls import re_path

from camac.gis import views

urlpatterns = [
    re_path(
        r"^data/(?:(?P<task_id>[\w-]+)/)?$",
        views.GISDataView.as_view(),
        name="gis-data",
    ),
    re_path(
        r"^apply$",
        views.GISApplyView.as_view(),
        name="gis-apply",
    ),
]
