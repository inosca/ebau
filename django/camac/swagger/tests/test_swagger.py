import pytest
from django.conf import settings
from django.urls import reverse
from pytest_lazy_fixtures import lf
from rest_framework import status


@pytest.mark.parametrize(
    "appconfig,_ech0211_settings",
    [
        (
            lf("set_application_be"),
            lf("be_ech0211_settings"),
        ),
        (
            lf("set_application_sz"),
            lf("sz_ech0211_settings"),
        ),
        (
            lf("set_application_so"),
            lf("so_ech0211_settings"),
        ),
        # for coverage of the actual url config we need to run a test that
        # does not override the actual url settings:
        ("kt_bern", lf("ech0211_settings")),
    ],
)
def test_swagger_schema(
    db,
    user,
    admin_client,
    settings,
    _ech0211_settings,
    appconfig,
    caplog,
    reload_ech0211_urls,
):
    response = admin_client.get(reverse("schema-json", args=[".json"]))
    assert response.status_code == status.HTTP_200_OK
    assert not len(caplog.messages)


@pytest.mark.parametrize("application_name", settings.APPLICATIONS.keys())
def test_swagger_paths(
    admin_client,
    application_name,
    settings,
    snapshot,
    application_settings,
    request,
):
    short_name = settings.APPLICATIONS[application_name]["SHORT_NAME"]

    request.getfixturevalue(f"set_application_{short_name}")
    try:
        request.getfixturevalue(f"{short_name}_ech0211_settings")
    except pytest.FixtureLookupError:
        request.getfixturevalue("disable_ech0211_settings")

    request.getfixturevalue("reload_ech0211_urls")
    response = admin_client.get(reverse("schema-json", args=[".json"]))
    result = response.json()

    assert sorted(set(result["paths"].keys())) == snapshot
