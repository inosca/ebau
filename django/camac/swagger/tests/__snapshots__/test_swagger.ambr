# serializer version: 1
# name: test_swagger_paths[demo]
  list([
  ])
# ---
# name: test_swagger_paths[kt_bern]
  list([
    '/api/v1/attachment-sections',
    '/api/v1/attachment-sections/{attachment_section_id}',
    '/api/v1/attachments',
    '/api/v1/attachments/files/',
    '/api/v1/attachments/{attachment_id}',
    '/api/v1/groups',
    '/api/v1/groups/{group_id}',
    '/api/v1/instances/{instance_id}/generate-pdf',
    '/api/v1/me',
    '/api/v1/public-services',
    '/api/v1/public-services/{service_id}',
    '/ech/v1/application/{instance_id}/',
    '/ech/v1/applications',
    '/ech/v1/message/',
    '/ech/v1/send/',
  ])
# ---
# name: test_swagger_paths[kt_gr]
  list([
    '/api/v1/groups',
    '/api/v1/groups/{group_id}',
    '/api/v1/me',
    '/api/v1/public-services',
    '/api/v1/public-services/{service_id}',
    '/ech/v1/application/{instance_id}/',
    '/ech/v1/applications',
    '/ech/v1/message/',
    '/ech/v1/send/',
  ])
# ---
# name: test_swagger_paths[kt_schwyz]
  list([
    '/api/v1/attachments',
    '/api/v1/attachments/files/',
    '/api/v1/attachments/{attachment_id}',
    '/api/v1/groups',
    '/api/v1/groups/{group_id}',
    '/api/v1/me',
    '/ech/v1/application/{instance_id}/',
    '/ech/v1/applications',
    '/ech/v1/message/',
  ])
# ---
# name: test_swagger_paths[kt_so]
  list([
  ])
# ---
# name: test_swagger_paths[kt_uri]
  list([
    '/public-caluma-instances',
  ])
# ---
# name: test_swagger_paths[test]
  list([
  ])
# ---
